ddtrace/profiling/collector/memalloc.py

```diff
@@ -58,7 +58,9 @@ class MemoryCollector(collector.PeriodicCollector):
     heap_sample_size = attr.ib(
         factory=attr_utils.from_env("DD_PROFILING_HEAP_SAMPLE_SIZE", _DEFAULT_HEAP_SAMPLE_SIZE, int)
     )
-    ignore_profiler = attr.ib(factory=attr_utils.from_env("DD_PROFILING_IGNORE_PROFILER", True, formats.asbool))
+    ignore_profiler = attr.ib(
+        factory=attr_utils.from_env("DD_PROFILING_IGNORE_PROFILER", True, formats.asbool)  # type: ignore[arg-type]
```

Comment:
This is due to the fact that `formats.asbool` is not an instance of `Type`, like in all other cases (e.g. `float`)
~~~
error: Argument 3 to "from_env" has incompatible type "Callable[[Union[str, bool, None]], bool]"; expected "Type[bool]"  [arg-type]
~~~
This could be fixed by allowing both `Type` and `Callable` in `from_env` I think

Reply:
Yes, sounds like a `from_env` typing issue we should fix